sdks/python/apache_beam/io/iobase.py

```diff
@@ -1618,3 +1628,48 @@ def display_data(self):
         'source': DisplayDataItem(self.source.__class__, label='Read Source'),
         'source_dd': self.source
     }
+
+
+class SDFBoundedSourceReader(PTransform):
```

Comment:
It seems like the major difference between `SDFBoundedSourceWrapper ` and `SDFBoundedSourceReader ` is that `SDFBoundedSourceWrapper` takes the source as construction param where `SDFBoundedSourceReader` takes the source as input element. We could change the implementation of `SDFBoundedSourceWrapper` as well.

Reply:
I've done this - but I've still allowed the source to come in via the constructor as well as as an input. The intention of doing this is to keep the display data for simple Read transforms where the source is known at construction time.

Reply:
I see. I thought we still keep `_SDFBoundedSourceWrapper `. Thanks for the clarification!

Reply:
I'm thinking whether it would be better for `SDFBoundedSourceReader ` to take data_to_display as constructor instead of source directly if any. What do you think?